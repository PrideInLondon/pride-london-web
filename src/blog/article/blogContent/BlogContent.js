import { documentToReactComponents } from '@contentful/rich-text-react-renderer'
import renderMethods from './renderMethods'

export const BlogContent = ({ article }) =>
  documentToReactComponents(article.json, renderMethods)
