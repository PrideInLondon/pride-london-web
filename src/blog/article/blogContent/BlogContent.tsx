import React from 'react'
import { documentToReactComponents } from '@contentful/rich-text-react-renderer'
import { PageContentProps } from '../PageContent.types'
import renderMethods from './renderMethods'

export const BlogContent: React.FC<Pick<PageContentProps, 'article'>> = ({
  article,
}) => <>{documentToReactComponents(article.json, renderMethods)}</>
