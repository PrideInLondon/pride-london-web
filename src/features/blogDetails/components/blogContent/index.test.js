import React from 'react'
import { shallow } from 'enzyme'
import BlogContent from '.'

describe(BlogContent.name, () => {
  it('renders with default values', () => {
    const article = {
      json: {
        nodeType: 'document',
        data: {},
        content: [
          {
            nodeType: 'paragraph',
            data: {},
            content: [
              {
                nodeType: 'text',
                value: 'Hello world!',
                marks: [],
                data: {},
              },
            ],
          },
        ],
      },
    }
    const wrapper = shallow(<BlogContent article={article} />)
    expect(wrapper).toMatchSnapshot()
  })
})
