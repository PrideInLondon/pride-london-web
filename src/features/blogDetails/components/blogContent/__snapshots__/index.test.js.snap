// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogContent renders with default values 1`] = `
<Paragraph
  key="0"
>
  Hello world!
</Paragraph>
`;
